tests/Bridge/Elasticsearch/DataProvider/PaginatorTest.php

```diff
@@ -149,20 +168,31 @@ function (array $document): Foo {
 
     protected function setUp()
     {
-        $this->paginator = $this->getPaginator(4, 4);
+        $this->paginator = $this->getPaginator($this->limit, $this->offset);
     }
 
     protected function getPaginator(int $limit, int $offset)
+    {
+        $denormalizerProphecy = $this->createDenormalizerProphecy(static::DOCUMENTS);
+        return $this->createPaginator($limit, $offset, self::DOCUMENTS, $denormalizerProphecy);
+    }
+
+    protected function createDenormalizerProphecy($documents)
     {
         $denormalizerProphecy = $this->prophesize(DenormalizerInterface::class);
 
-        foreach (static::DOCUMENTS['hits']['hits'] as $document) {
+        foreach ($documents['hits']['hits'] as $document) {
             $denormalizerProphecy
                 ->denormalize($document, Foo::class, ItemNormalizer::FORMAT, [AbstractNormalizer::ALLOW_EXTRA_ATTRIBUTES => true])
                 ->willReturn($this->denormalizeFoo($document['_source']));
         }
+        return $denormalizerProphecy;
+    }
 
-        return new Paginator($denormalizerProphecy->reveal(), self::DOCUMENTS, Foo::class, $limit, $offset);
+    protected function createPaginator(int $limit, int $offset, $documents, $denormalizerProphecy)
```

Comment:
Just copy/paste this line to keep the tests explicit.

Reply:
I remove createPaginator function , and ai use new Paginator explicit in the tests.